### ISSUE TITLE
Touchpad not working Lenovo Yoga 720 w/Deepin Beta 20

### ISSUE THREAD

**Issue by chaim1221** [linuxdeepin/developer-center#1905]
## Describe the bug
Some time after install, the touchpad stopped working.

## To Reproduce
1. Have a Yoga 720 computer.
2. Perform a normal install (see issue #1903).
3. Connect other devices such as bluetooth mouse. Drawing pad is set to pen (this doesn't seem to matter, but maybe). 
4. Eventually the touchpad will stop working.

## Expected behavior
The touchpad is a critical component and should never stop working in an X window system!

### Distro & Version
Linux Deepin Beta 20
```
celiyah@ce-y720-dev:~$ uname -r
5.3.0-3-amd64
```

There is a thread [on Reddit](https://www.reddit.com/r/deepin/comments/hlo7q6/deepin_20_beta_158_trackpad_not_working_lenovo/) but the OP never followed up. You can see me trying the one suggestion there.

**Comment by xmuli** [comment]
> [Reddit上](https://www.reddit.com/r/deepin/comments/hlo7q6/deepin_20_beta_158_trackpad_not_working_lenovo/)有一个线程[，](https://www.reddit.com/r/deepin/comments/hlo7q6/deepin_20_beta_158_trackpad_not_working_lenovo/)但是OP从未跟进。

After installing this name, is it still not found?   [acpidump](https://ilinuxkernel.com/?p=2001)

**Comment by chaim1221** [comment]
```
celiyah@ce-y720-dev:~$ sudo acpidump | grep -C3 ELAN
    ECD0: 41 54 4D 4C 32 39 35 32 00 5F 48 49 44 70 00 48  ATML2952._HIDp.H
    ECE0: 49 44 32 70 0A 4A 42 41 44 52 70 0C 80 1A 06 00  ID2p.JBADRp.....
    ECF0: 53 50 45 44 A4 00 A0 30 93 53 44 53 31 0A 03 70  SPED...0.SDS1..p
    ED00: 0D 45 4C 41 4E 32 30 39 37 00 5F 48 49 44 70 01  .ELAN2097._HIDp.
    ED10: 48 49 44 32 70 0A 10 42 41 44 52 70 0C 80 1A 06  HID2p..BADRp....
    ED20: 00 53 50 45 44 A4 00 A0 30 93 53 44 53 31 0A 04  .SPED...0.SDS1..
    ED30: 70 0D 4E 54 52 47 30 30 30 31 00 5F 48 49 44 70  p.NTRG0001._HIDp
--
   19140: 49 30 4C 50 43 42 5B 82 47 1D 50 53 32 4D 08 54  I0LPCB[.G.PS2M.T
   19150: 50 49 44 12 4D 06 04 12 1B 05 0A 04 0A 2C 0A 20  PID.M........,. 
   19160: 0D 53 59 4E 41 32 42 32 43 00 0D 50 4E 50 30 43  .SYNA2B2C..PNP0C
   19170: 35 30 00 12 1A 05 0A 08 0A 15 01 0D 45 4C 41 4E  50..........ELAN
   19180: 30 36 35 31 00 0D 50 4E 50 30 43 35 30 00 12 1A  0651..PNP0C50...
   19190: 05 0A FE 0A 15 01 0D 45 4C 41 4E 30 36 35 31 00  .......ELAN0651.
   191A0: 0D 50 4E 50 30 43 35 30 00 12 17 05 0A FF 0A FF  .PNP0C50........
   191B0: 0A FF 0D 4D 53 46 54 30 30 30 33 00 0C 41 D0 0F  ...MSFT0003..A..
   191C0: 13 14 4E 05 5F 48 49 44 08 08 5F 54 5F 30 00 A0  ..N._HID.._T_0..
--
   19360: 2C 0A 20 0D 53 59 4E 41 32 42 32 43 00 0D 50 4E  ,. .SYNA2B2C..PN
   19370: 50 30 43 35 30 00 12 1A 05 0A 08 0A 15 01 0D 45  P0C50..........E
   19380: 4C 41 4E 30 36 35 31 00 0D 50 4E 50 30 43 35 30  LAN0651..PNP0C50
   19390: 00 12 1A 05 0A FE 0A 15 01 0D 45 4C 41 4E 30 36  ..........ELAN06
   193A0: 35 31 00 0D 50 4E 50 30 43 35 30 00 12 17 05 0A  51..PNP0C50.....
   193B0: FF 0A FF 0A FF 0D 4D 53 46 54 30 30 30 33 00 0C  ......MSFT0003..
   193C0: 41 D0 0F 13 08 53 42 46 42 11 26 0A 23 8E 1E 00  A....SBFB.&.#...
```

Of course! It's been a long day.

**Comment by chaim1221** [comment]
Any advice? Other than telling me that it's probably actually a Synaptic touchpad (no surprise), the above hex-ascii table does not mean much to me.

**Comment by chaim1221** [comment]
There's new info in Reddit. So it's detecting the touchpad, it's just not using it. :-/

**Comment by chaim1221** [comment]
Notable updates from the Reddit thread
- Theirs is an AMD Ryzen problem, mine is an Intel machine (I had _no_ idea the touchpads were different).
- Here's the shakedown on what the acpi output means:
```
ELAN2097 is the touchscreen
ELAN0651 is the touchpad
```
- The touchscreen is working, the touchpad isn't
- I'm going to look for known driver issues related to the touchpad.

**Comment by xmuli** [comment]
I don’t know much about driver issues,   I don’t know if there is any reference on Arch Wiki (you can check it out)

**Comment by chaim1221** [comment]
I don't think Arch will help much in our case? This is UOS. I can check for Debian. But check this out:
```
celiyah@ce-y720-dev:~$ sudo dmesg | grep -i elan
[    1.555110] i2c_hid i2c-ELAN0651:00: i2c-ELAN0651:00 supply vdd not found, using dummy regulator
[    1.555129] i2c_hid i2c-ELAN0651:00: i2c-ELAN0651:00 supply vddl not found, using dummy regulator
[    1.665739] input: ELAN0651:00 04F3:3052 Mouse as /devices/pci0000:00/0000:00:15.2/i2c_designware.2/i2c-8/i2c-ELAN0651:00/0018:04F3:3052.0003/input/input11
[    1.665928] input: ELAN0651:00 04F3:3052 Touchpad as /devices/pci0000:00/0000:00:15.2/i2c_designware.2/i2c-8/i2c-ELAN0651:00/0018:04F3:3052.0003/input/input12
[    1.666122] hid-generic 0018:04F3:3052.0003: input,hidraw2: I2C HID v1.00 Mouse [ELAN0651:00 04F3:3052] on i2c-ELAN0651:00
[   13.537883] input: ELAN0651:00 04F3:3052 Touchpad as /devices/pci0000:00/0000:00:15.2/i2c_designware.2/i2c-8/i2c-ELAN0651:00/0018:04F3:3052.0003/input/input19
[   13.538112] hid-multitouch 0018:04F3:3052.0003: input,hidraw2: I2C HID v1.00 Mouse [ELAN0651:00 04F3:3052] on i2c-ELAN0651:00
```

```
celiyah@ce-y720-dev:~$ sudo lsmod | grep intel_lpss_pci
intel_lpss_pci         20480  0
intel_lpss             16384  1 intel_lpss_pci
```

It's _definitely_ detecting it and the driver is loaded.

**Comment by peeweep** [comment]
If you still have this problem on the latest version of Deepin, try configuring `libinput` or `udev`.

**Comment by babyfengfjx** [comment]
Hey, I don't know if your problem has been solved in the latest Deepin V20.4 version?

**Comment by BLumia** [comment]
Closed since no further response. If you still have this problem on the latest version of deepin (at the time of writing we are on deepin v20.5 now), please feel free to create a new issue ticket to report the issue so we can look into it. Thanks!
